core/solver/bicgstab.cpp

```diff
@@ -90,7 +91,7 @@ std::unique_ptr<LinOp> Bicgstab<ValueType>::conj_transpose() const
 template <typename ValueType>
 void Bicgstab<ValueType>::apply_impl(const LinOp* b, LinOp* x) const
 {
-    precision_dispatch_real_complex<ValueType>(
+    precision_dispatch_real_complex_distributed<ValueType>(
```

Comment:
I'm not sure if we run into issues at unexpected places if we do that switch. Currently, distriuted is not the default and I think it would require more work to make it the default. For now, I would leave it in this explicit state.